### ISSUE TITLE
Provide JSX support

### ISSUE THREAD

**Issue by edgarmueller** [eclipsesource/jsonforms#629]
Renderers should be written in JSX instead of the native DOM API.

**Comment by eneufeld** [comment]
I looked for alternatives and arguments pro and contra JSX.
ES6 added "Template Strings" Here are some descriptions:
- [ES6 Tagged Template Strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) 
- [2nd Description](http://exploringjs.com/es6/ch_template-literals.html#sec_tagged-templates-examples)
Here is the argument by facebook why they created JSX instead of using tagged template strings:
- [JSX vs Template Strings](https://facebook.github.io/jsx/#why-not-template-literals)

There are also libs to convert template strings to virtual dom:
- [HyperX](https://github.com/substack/hyperx)
- [T7](https://github.com/trueadm/t7)

I think we must evaluate whether we can mix JSX and "plain" node creation.
I also like the tagged template strings

**Comment by edgarmueller** [comment]
[Stencil](https://stenciljs.com/) might also be worth a look

**Comment by eneufeld** [comment]
I looked at the whole VirtualDom thing. Here are some of my thoughts:
A positive aspect of a VirtualDom are ease to write Dynamic Components as you don't need dynamic code, as dynamic changes will be diffed by the library and thus propagated to the DOM.
On the otherside there is no common implementation of a VirtualDom. So the use of a VirtualDOM leads to a dependency on the library. A change of the library is expensive.
If we don't use JSX and use simple template strings then we can create easy static components. But the dynamic components will still be complicated (see the current Tree or Table renderer).
If we decide to go with a VirtualDOM, then we must look at all available implementations.
I found this benchmark: [Benchmark](https://github.com/krausest/js-framework-benchmark) .

If we could abstract our dependency to a VirtualDOM library I would be all for it.
Currently I'm undecided.
As the main question I have is, what kind of renderers do we want to make easy to write. If it is only the simple ones, I would go with the template strings. If we want also make "complicated" renderers easy to write then a VirtualDOM is the way to go.
